src/external/interbtc-index.ts

```diff
@@ -0,0 +1,53 @@
+import * as interbtcIndex from "@interlay/interbtc-index-client";
+
+// The generated client contains the following autogenerated middleware helpers, which must be filtered out
+const GeneratedMiddlewareFns = ["withMiddleware", "withPreMiddleware", "withPostMiddleware"] as const;
+type GeneratedMiddlewareFns = typeof GeneratedMiddlewareFns[number];
+// For every 'foo() => Promise<T>' function, the generated client
+// has a 'fooRaw() => <Promise<ApiResponse<T>>' counterpart. These must be filtered out
+type RawApiResponse = Promise<interbtcIndex.ApiResponse<any>>;
+
+export type IndexAPI = Pick<
+    interbtcIndex.IndexApi,
+    {
+        [ApiFn in keyof interbtcIndex.IndexApi]-?: ReturnType<interbtcIndex.IndexApi[ApiFn]> extends RawApiResponse
```

Comment:
Just for my understanding: Just using the `Required<T>` type here would not have been enough, because what you're filtering out with `-?` are the `never`s generated in the lines below?

Reply:
Yep, correct